### PR TITLE
Full RPC rewrite with threading

### PR DESCRIPTION
**Pull request by zpalmtree**
A full RPC rewrite. Runs the RPC on a separate thread.

TODO: Synchronize data races, testing

Resolves: #603

### CODE REVIEW

**Comment by ExtraHash** [comment]
Wow, cool PR

**Comment by kryptonchain** [comment]
Well done.

**Comment by brandonlehmann** [comment]
I'm running into compilation issues on windows after that latest push and it appears that appveyor is as well. 

My build was from https://github.com/turtlecoin/turtlecoin/pull/903/commits/2dd48fcdc0ee4e4ebf465f8372f1f1afe5f96a67 last night and I had a lot of exceptions thrown. This is definitely a good idea and a step in the right direction but we need to work to avoid throwing and avoiding race conditions as much as possible.

**Comment by zpalmtree** [comment]
> I'm running into compilation issues on windows after that latest push and it appears that appveyor is as well.
> 
> My build was from [2dd48fc](https://github.com/turtlecoin/turtlecoin/commit/2dd48fcdc0ee4e4ebf465f8372f1f1afe5f96a67) last night and I had a lot of exceptions thrown. This is definitely a good idea and a step in the right direction but we need to work to avoid throwing and avoiding race conditions as much as possible.

Latest commit should fix windows compilation issues. I am going to look more into the race conditions / crashes but am not exactly sure how we're going to manage to selectively synchronize stuff in core without blocking the whole thing.

**Comment by brandonlehmann** [comment]
> I am going to look more into the race conditions / crashes but am not exactly sure how we're going to manage to selectively synchronize stuff in core without blocking the whole thing.

I hear ya. I'd definitely like to run this under load for a bit longer before this gets merged as the risk may outweigh the benefit in some circumstances.

**Comment by zpalmtree** [comment]
I think there's something wrong with /getwalletsyncdata with this change, not seeing it getting hit in the logs via a wallet, but I am seeing it when manually curling it.. will investigate later.

**Comment by LeoStehlik** [comment]
definitely a step into the right direction @zpalmtree thank you for this! I will test wallet sync later, but when curling getwalletsyncdata against (dego) daemon at .1sec cadence, ... even with 0 txs in the pool, I never seen the daemon respond so fast and consistently. Sure it will take time to catch the lose ends, race conditions bugs it may introduce, but I like this already!

**Comment by zpalmtree** [comment]
Full rewrite can be found here: https://github.com/turtlecoin/turtlecoin/compare/development...zpalmtree:threaded-rpc-wip?expand=1

I am not pushing it up to this PR yet as it would break people building from this branch. The branch linked above now has enough endpoints to sync and send transactions with zedwallet-beta.

**Comment by zpalmtree** [comment]
Full rewrite progress:

### Conventional handlers
- [ ] `/getinfo` - Not going to implement
- [ ] `/getheight` - Not going to implement
- [ ] `/feeinfo` - Not going to implement
- [ ] `/getpeers` - Not going to implement
- [x] `/info`
- [x] `/height`
- [x] `/fee`
- [x] `/peers`
- [ ] `/gettransactions`
- [x] `/sendrawtransaction`
- [ ] `/getblocks`
- [ ] `/queryblocks`
- [ ] `/queryblockslite`
- [ ] `/queryblocksdetailed`
- [x] `/getwalletsyncdata`
- [ ] `/get_o_indexes`
- [x] `/getrandom_outs`
- [ ] `/get_pool_changes`
- [ ] `/get_pool_changes_lite`
- [ ] `/get_block_details_by_height`
- [ ] `/get_blocks_details_by_heights`
- [ ] `/get_blocks_details_by_hashes`
- [ ] `/get_blocks_hashes_by_timestamps`
- [ ] `/get_transaction_details_by_hashes`
- [ ] `/get_transaction_hashes_by_payment_id`
- [x] `/get_global_indexes_for_range`
- [ ] `/get_transaction_status`

### Json RPC handlers

- [ ] `/f_blocks_list_json`
- [ ] `/f_block_json`
- [ ] `/f_transaction_json`
- [ ] `/f_on_transactions_pool_json`
- [x] `/getblockcount`
- [x] `/on_getblockhash`
- [x] `/getblocktemplate`
- [ ] `/getcurrencyid`
- [x] `/submitblock`
- [x] `/getlastblockheader`
- [x] `/getblockheaderbyhash`
- [ ] `/getblockheaderbyheight`

**Comment by brandonlehmann** [comment]
I've crossed out the ones I would nix (even if you said you aren't going to implement (sorry for potato formatting)

In short, if it doesn't facilitate information about the node (info, height, fee, peers), does not power wallet syncing (either legacy or Nigel), or can be used to build a block explorer, it needs to go.

### Conventional handlers
* [ ]  ~~`/getinfo` - Not going to implement~~
* [ ]  ~~`/getheight` - Not going to implement~~
* [ ]  ~~`/feeinfo` - Not going to implement~~
* [ ]  ~~`/getpeers` - Not going to implement~~
* [x]  `/info`
* [x]  `/height`
* [x]  `/fee`
* [x]  `/peers`
* [ ]  ~~`/gettransactions`~~
* [x]  `/sendrawtransaction`
* [ ]  ~~`/getblocks`~~
* [ ] ~~`/queryblocks`~~
* [ ]  `/queryblocksdetailed` *needed for foreseeable future, please don't nerf yet :)*
* [ ]  `/queryblockslite` **required for NodeRpcProxy (until turtle-service is dead)**
* [x]  `/getwalletsyncdata`
* [ ] ~~`/get_o_indexes`~~ **remove from NodeRpcProxy**
* [ ] ~~`/get_blocks_details_by_hashes` - Perhaps should be removed?~~ **remove from NodeRpcProxy**
* [x]  `/getrandom_outs`
* [ ]  ~~`/get_pool_changes`~~
* [ ]  `/get_pool_changes_lite`
* [ ]  ~~`/get_block_details_by_height`~~ **remove from NodeRpcProxy**
* [ ]  ~~`/get_blocks_details_by_heights` - Perhaps should be removed?~~ **remove from
NodeRpcProxy**
* [ ]  ~~`/get_blocks_details_by_hashes` - Perhaps should be removed?~~ **remove from
NodeRpcProxy**
* [ ]  ~~`/get_blocks_hashes_by_timestamps` - Perhaps should be removed?~~ **remove from
NodeRpcProxy**
* [ ]  ~~`/get_transaction_details_by_hashes` - Perhaps should be removed?~~ **remove from
NodeRpcProxy**
* [ ]  ~~`/get_transaction_hashes_by_payment_id` - Perhaps should be removed?~~ **remove from
NodeRpcProxy**
* [x]  `/get_global_indexes_for_range`
* [ ]  ~~`/get_transaction_status`~~ *I can't find this one*

### Json RPC handlers
* [ ]  `/f_blocks_list_json`
* [ ]  `/f_block_json`
* [ ]  `/f_transaction_json`
* [ ]  `/f_on_transactions_pool_json`
* [x]  `/getblockcount`
* [x]  ~~`/on_getblockhash`~~ **this can be handled by /getblockheaderbyheight**
* [x]  `/getblocktemplate`
* [ ]  ~~`/getcurrencyid` - Perhaps should be removed?~~
* [x]  `/submitblock`
* [x]  `/getlastblockheader`
* [x]  `/getblockheaderbyhash`
* [x]  `/getblockheaderbyheight`

**Comment by zpalmtree** [comment]
>I can't find this one
https://github.com/turtlecoin/turtlecoin/blob/development/src/nigel/Nigel.cpp#L332

**Comment by brandonlehmann** [comment]
> > I can't find this one
> 
> https://github.com/turtlecoin/turtlecoin/blob/development/src/nigel/Nigel.cpp#L332

That's `get_transactions_status` -- the `s` is a killer

**Comment by zpalmtree** [comment]
Pushed up the full RPC rewrite, with all methods completed that I plan to implement.

I now need to cleanup the NodeRpcProxy code, and figure out how to synchronize access to Core to fix rare crashes.

**Comment by zpalmtree** [comment]
Legacy blockchain explorer has been removed. NodeRpcProxy code has been cleaned up.

**Comment by zpalmtree** [comment]
This is ready for review and testing. I need to do more testing to try and reproduce crashes, but people helping me with this would not hurt. I didn't manage to reproduce the previous crashes using /info, but have not got around to testing other endpoints.

**Comment by brandonlehmann** [comment]
> This is ready for review and testing. I need to do more testing to try and reproduce crashes, but people helping me with this would not hurt. I didn't manage to reproduce the previous crashes using /info, but have not got around to testing other endpoints.

Roger that. Looks like a bit of stuff to read through :)

File: src/rpc/RpcServer.cpp [hunk 1]
```diff
@@ -1,1579 +1,3055 @@
...
+        if (!hasMember(*body, "method"))
+        {
+            failRequest(400, "Missing JSON parameter: 'method'", res);
+            return;
+        }
+
+        const auto method = getStringFromJSON(*body, "method");
 
-#undef ERROR
+        if (method == "getblocktemplate")
```
Comment (brandonlehmann) [inline src/rpc/RpcServer.cpp]: What's wrong with a switch statement?

File: src/rpc/RpcServer.cpp [hunk 2]
```diff
@@ -1,1579 +1,3055 @@
...
+            .Post("/queryblockslite", router(&RpcServer::queryBlocksLite, RpcMode::Default, bodyRequired))
+            .Post("/get_transactions_status", router(&RpcServer::getTransactionsStatus, RpcMode::Default, bodyRequired))
+            .Post("/get_pool_changes_lite", router(&RpcServer::getPoolChanges, RpcMode::Default, bodyRequired))
+            .Post("/queryblocksdetailed", router(&RpcServer::queryBlocksDetailed, RpcMode::AllMethodsEnabled, bodyRequired))
+            .Post("/get_o_indexes", router(&RpcServer::getGlobalIndexesDeprecated, RpcMode::Default, bodyRequired))
+
+            /* Matches everything */
+            /* NOTE: Not passing through middleware */
+            .Options(".*", [this](auto &req, auto &res) { handleOptions(req, res); });
+}
```
Comment (brandonlehmann) [inline src/rpc/RpcServer.cpp]: Does the lib auto kick back a 404 on undefined paths?

File: src/rpc/RpcServer.cpp [hunk 3]
```diff
@@ -1,1579 +1,3055 @@
...
+        "Incoming " + req.method + " request: " + req.path,
+        Logger::DEBUG,
+        { Logger::DAEMON_RPC }
+    );
+
+    if (m_corsHeader != "")
     {
-        s(keyOutput.key, "key");
-        s(keyOutput.amount, "amount");
+        res.set_header("Access-Control-Allow-Origin", m_corsHeader);
```
Comment (brandonlehmann) [inline src/rpc/RpcServer.cpp]: Should we put the other preflight headers in here (for non-smart clients)?

File: src/rpc/RpcServer.cpp [hunk 4]
```diff
@@ -1,1579 +1,3055 @@
...
-        return m_cors_domains;
+        res.set_header("Allow", supported);
     }
 
-    bool RpcServer::isCoreReady()
+    if (m_corsHeader != "")
     {
-        return m_p2p.get_payload_object().isSynchronized();
+        res.set_header("Access-Control-Allow-Origin", m_corsHeader);
+        res.set_header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept, X-API-KEY");
```
Comment (brandonlehmann) [inline src/rpc/RpcServer.cpp]: X-API-KEY not needed here

File: src/rpc/RpcServer.cpp [hunk 5]
```diff
@@ -1,1579 +1,3055 @@
...
-        res.status = CORE_RPC_STATUS_OK;
+    writer.Key("status");
+    writer.String("OK");
 
-        std::stringstream ss;
+    writer.EndObject();
 
-        std::for_each(res.outs.begin(), res.outs.end(), [&](auto &ofa) {
-            ss << "[" << ofa.amount << "]:";
+    res.set_content(sb.GetString(), "application/json");
```
Comment (brandonlehmann) [inline src/rpc/RpcServer.cpp]: Needing to set the content type like this seems repetitive. Is there a better way to handle this?

File: src/cryptonoteprotocol/CryptoNoteProtocolHandler.cpp [hunk 6]
```diff
@@ -54,7 +54,7 @@ namespace CryptoNote
 
             for (const auto &rawBlock : rawBlocks)
             {
-                legacy.emplace_back(RawBlockLegacy {rawBlock.block, rawBlock.transactions});
+                legacy.emplace_back(rawBlock.block, rawBlock.transactions);
```
[the remark this answers is not in the document]
Comment (zpalmtree) [reply src/cryptonoteprotocol/CryptoNoteProtocolHandler.cpp]: That's a `RawBlock`, not a `RawBlockLegacy`. `RawBlock` does not have a constructor, so we can use aggregate initialization. I added a constructor to `RawBlockLegacy`, so we can no longer use aggregate initialization.

Therefore, cannot update L70 without also giving it a constructor.

File: src/rpc/RpcServer.cpp (hunk 2 above)
Comment (zpalmtree) [reply src/rpc/RpcServer.cpp]: Yes

File: src/cryptonoteprotocol/CryptoNoteProtocolHandler.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (brandonlehmann) [reply src/cryptonoteprotocol/CryptoNoteProtocolHandler.cpp]: So what you’re saying is you’re half done?

File: src/rpc/RpcServer.cpp (hunk 3 above)
Comment (zpalmtree) [reply src/rpc/RpcServer.cpp]: I admit to being somewhat of a novice on handling CORS, but from my knowledge we only need to add these in the Options request which sends the preflight requests. If you could point to a source indicating we should include them in normal requests as well, I'll happily update.

File: src/rpc/RpcServer.cpp (hunk 4 above)
Comment (zpalmtree) [reply src/rpc/RpcServer.cpp]: This is true, but I have designed this rewrite to make it very easy to extend in the future, and I could imagine we may at some point desire endpoints that require authentication.

File: src/rpc/RpcServer.cpp (hunk 5 above)
Comment (zpalmtree) [reply src/rpc/RpcServer.cpp]: Yes. I will update shortly

File: src/cryptonoteprotocol/CryptoNoteProtocolHandler.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (zpalmtree) [reply src/cryptonoteprotocol/CryptoNoteProtocolHandler.cpp]: I don't see any reason to touch unrelated code adding a constructor for something which doesn't need one

File: src/rpc/RpcServer.cpp (hunk 3 above)
Comment (brandonlehmann) [reply src/rpc/RpcServer.cpp]: It’s overkill.

**Comment by zpalmtree** [comment]
>I was quite surprised to see all of the JSON serialization living in RpcServer.cpp -- are those really one-time use for just the RPC like that or is there a case for keeping them elsewhere like previously?
I could imagine in the future we *may* want the JSON serialization to be used elsewhere. However, until that day arrives, I would much rather keep it here in the RpcServer.

My reasoning is as follows:
* The current serialization code is very hard to follow, and very far away from where it is used
* We do not yet need this serialization elsewhere
* We won't save many lines of code by pulling out some of the duplication into methods
* By the majority of methods in the RPC being simply making a core call, then looping through the data and writing it, it is both very simple to read and follow, and also super easy to work out the data structure returned without having to manually run the calls, some of which have conditional data.

File: src/rpc/RpcServer.cpp (hunk 4 above)
Comment (brandonlehmann) [reply src/rpc/RpcServer.cpp]: I would argue that if such functionality is added later that requires it that adding the extra header here would be part of that addition. Otherwise, if I saw this in a response I would expect that something is available that can use it.

Comment (zpalmtree) [reply src/rpc/RpcServer.cpp]: OK, will update shortly

File: src/rpc/RpcServer.cpp (hunk 5 above)
Comment (zpalmtree) [reply src/rpc/RpcServer.cpp]: Done

File: src/rpc/RpcServer.cpp (hunk 4 above)
Comment (zpalmtree) [reply src/rpc/RpcServer.cpp]: Done

**Comment by brandonlehmann** [comment]
Does this solve #603 in the process?

**Comment by zpalmtree** [comment]
> Does this solve #603 in the process?

Almost certainly. I will verify later today.

**Comment by zpalmtree** [comment]
Tested - Yep, it fixes it.

**Comment by brandonlehmann** [comment]
`/getblocks` returns the raw blocks. On second thought, can we keep that one too?

**Comment by zpalmtree** [comment]
> `/getblocks` returns the raw blocks. On second thought, can we keep that one too?

OK, will update

**Comment by zpalmtree** [comment]
Also, does anything use '/getblocks' right now? I wouldn't mind modernizing it if not since I plan to use it.

**Comment by brandonlehmann** [comment]
Nothing yet that I’m aware of. Feel free to update it a bit.

**Comment by zpalmtree** [comment]
Added `/getrawblocks`, functions nearly identically to `/getwalletsyncdata`, only returning the raw block + transactions in hex. Using a new endpoint to avoid confusion when communicating with both old and new daemons.

**Comment by brandonlehmann** [comment]
> Added `/getrawblocks`, functions nearly identically to `/getwalletsyncdata`, only returning the raw block + transactions in hex. Using a new endpoint to avoid confusion when communicating with both old and new daemons.

Thanks. Will check it out.

**Comment by zpalmtree** [comment]
I think there is an issue with `/get_global_indexes_for_range` not always returning the right data. Furthermore, I think there is a bug where if this fails, the whole sync process comes to a halt.

**Comment by zpalmtree** [comment]
Need to add user agent to request logging

**Comment by zpalmtree** [comment]
User agent added

**Comment by brandonlehmann** [comment]
Does this also adjust for #704 per chance? Or should #704 be worked separately?

**Comment by zpalmtree** [comment]
#704 is an issue in Core, not the RPC

**Comment by brandonlehmann** [comment]
> #704 is an issue in Core, not the RPC

Thanks for the clarification. Will proceed with checking #704 independently.

**Comment by zpalmtree** [comment]
roundToMidnight in the DB backends iirc

**Comment by zpalmtree** [comment]
Fixed failing to get indexes halting sync process. Fixed connections not getting properly reused.

**Comment by zpalmtree** [comment]
I think this is ready for review/testing again

File: external/cpp-httplib/httplib.h [hunk 7]
```diff
@@ -90,7 +90,7 @@ typedef int socket_t;
 /*
  * Configuration
  */
-#define CPPHTTPLIB_KEEPALIVE_TIMEOUT_SECOND 5
+#define CPPHTTPLIB_KEEPALIVE_TIMEOUT_SECOND 30
```
Comment (brandonlehmann) [inline external/cpp-httplib/httplib.h]: What's the logic behind this bump here? A public node may hurt a bit by holding connections open this long.

Comment (zpalmtree) [reply external/cpp-httplib/httplib.h]: Once we are fully synced, our connections won't remain open, since we hit /info every 5 or 10 seconds. This helps prevents us from spawning tons of connections.

Comment (brandonlehmann) [reply external/cpp-httplib/httplib.h]: Okay. Gotcha.

**Comment by brandonlehmann** [comment]
> I've been running this and it seems to be okay at this point.

Would like it if a pool op would test this PR while mining to verify it's solid on the block template and submit block sides of things.

**Comment by LeoStehlik** [comment]
testing on the fork coin atm. all looks good, getting blocks. will test on TRTL tomorrow.

**Comment by zpalmtree** [comment]
merge conflicts resolved

**Comment by bobbieltd** [comment]
I think this log is normal, still posting here just in case :
2019-Nov-27 09:47:00.823858 FATAL   [daemon] Caught unexpected exception: std::bad_alloc while processing /json_rpc request for User-Agent: 
2019-Nov-27 09:47:00.823987 FATAL   [daemon] Body: {"id":"0","jsonrpc":"2.0","method":"getlastblockheader","password":"","params":{},"timeout":360000}

**Comment by zpalmtree** [comment]
@bobbieltd Thanks, a few people had reported crashes but I wasn't sure what endpoint was causing them, I'll try and reproduce.

**Comment by zpalmtree** [comment]
Slightly more informative error message:
```
TurtleCoind: /home/gentoo/turtlecoin/turtlecoin/src/cryptonotecore/MemoryBlockchainStorage.cpp:28: virtual CryptoNote::RawBlock CryptoNote::MemoryBlockchainStorage::getBlockByIndex(uint32_t) const: Assertion `index < getBlockCount()' failed.
```

Stack trace:
```
#3  0x00007fdb71a90526 in __assert_fail () from /usr/lib/libc.so.6
#4  0x0000562af560df67 in CryptoNote::MemoryBlockchainStorage::getBlockByIndex(unsigned int) const ()
#5  0x0000562af56012e4 in CryptoNote::BlockchainStorage::getBlockByIndex(unsigned int) const ()
#6  0x0000562af55ae49d in CryptoNote::BlockchainCache::getBlockByIndex(unsigned int) const ()
#7  0x0000562af54c026e in CryptoNote::Core::restoreBlockTemplate(CryptoNote::IBlockchainCache*, unsigned int) const ()
#8  0x0000562af54d5ce5 in CryptoNote::Core::getBlockDetails(Crypto::Hash const&) const ()
#9  0x0000562af54d5b8f in CryptoNote::Core::getBlockDetails(unsigned int) const ()
#10 0x0000562af5930dec in RpcServer::info(httplib::Request const&, httplib::Response&, rapidjson::GenericDocument<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator>, rapidjson::CrtAllocator> const&) ()
```

**Comment by zpalmtree** [comment]
I think the keep alive forever modifications I made is preventing the RPC server from correctly shutting down. Probably just need to add a `shouldStop` var and check that in the loop.

**Comment by zpalmtree** [comment]
I believe I have fixed one of the most common crashes. Please let me know if this fixes the crashes for you, or you still get crashes from other methods.

**Comment by zpalmtree** [comment]
Another crash (probably only in debug mode)

```
TurtleCoind: /home/gentoo/turtlecoin/turtlecoin/src/cryptonotecore/Core.cpp:3650: CryptoNote::TransactionDetails CryptoNote::Core::getTransactionDetails(const Crypto::Hash &, CryptoNote::IBlockchainCache *, bool) const: Assertion `result == ExtractOutputKeysResult::SUCCESS' failed.
```

Stack trace:
```
#3  0x00007f4ac12d6526 in __assert_fail () from /usr/lib/libc.so.6
#4  0x000055e3317e7776 in CryptoNote::Core::getTransactionDetails(Crypto::Hash const&, CryptoNote::IBlockchainCache*, bool) const ()
#5  0x000055e3317e678c in CryptoNote::Core::getBlockDetails(Crypto::Hash const&) const ()
#6  0x000055e3317e5c46 in CryptoNote::Core::getBlockDetails(unsigned int, unsigned int) const ()
#7  0x000055e331c4150e in RpcServer::info(httplib::Request const&, httplib::Response&, rapidjson::GenericDocument<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator>, rapidjson::CrtAllocator> const&) ()
```

And as with the other crash, the core thread is in the middle of adding a block:

```
#26 0x000055e331843933 in CryptoNote::DatabaseBlockchainCache::getKeyOutputsCountForAmount(unsigned long, unsigned int) const ()
#27 0x000055e3318bccee in CryptoNote::BlockchainCache::insertKeyOutputToGlobalIndex(unsigned long, CryptoNote::PackedOutIndex, unsigned int) ()
#28 0x000055e3318badfd in CryptoNote::BlockchainCache::pushTransaction(CryptoNote::CachedTransaction const&, unsigned int, unsigned short) ()
#29 0x000055e3318ba623 in CryptoNote::BlockchainCache::doPushBlock(CryptoNote::CachedBlock const&, std::vector<CryptoNote::CachedTransaction, std::allocator<CryptoNote::CachedTransaction> > const&, CryptoNote::TransactionValidatorState const&, unsigned long, unsigned long, unsigned long, CryptoNote::RawBlock&&) ()
#30 0x000055e3318ba862 in CryptoNote::BlockchainCache::pushBlock(CryptoNote::CachedBlock const&, std::vector<CryptoNote::CachedTransaction, std::allocator<CryptoNote::CachedTransaction> > const&, CryptoNote::TransactionValidatorState const&, unsigned long, unsigned long, unsigned long, CryptoNote::RawBlock&&) ()
#31 0x000055e3317d7c58 in CryptoNote::Core::addBlock(CryptoNote::CachedBlock const&, CryptoNote::RawBlock&&) ()
#32 0x000055e3317db7af in CryptoNote::Core::addBlock(CryptoNote::RawBlock&&)
```

**Comment by zpalmtree** [comment]
> I think the keep alive forever modifications I made is preventing the RPC server from correctly shutting down. Probably just need to add a `shouldStop` var and check that in the loop.

Fixed

**Comment by brandonlehmann** [comment]
@zpalmtree sorry m8. Conflicts be afoot.

Any other reported issues on this lately?

**Comment by zpalmtree** [comment]
> @zpalmtree sorry m8. Conflicts be afoot.

fixed

> Any other reported issues on this lately?

Nope, not sure if anyone has ran the updated version after committing 7caca54

**Comment by brandonlehmann** [comment]
#yolo?

**Comment by zpalmtree** [comment]
> #yolo?

Hmm, I think there certainly are going to be a few crashes / bugs which are going to be very hard to track down, but I think the benefits might outweigh the negatives and more users will help us find the bugs.

**Comment by brandonlehmann** [comment]
giphy

**Comment by LeoStehlik** [comment]
I will run again (soon) for pool for both TurtleCoin and forked coin w. detailed logs to be able to report crashes. Big thanks for this @zpalmtree !
